### ISSUE TITLE
REST not working properly with SESSION in codeigniter 3.0

### ISSUE THREAD

**Issue by hkachhia** [chriskacerguis/codeigniter-restserver#456]
Hello,

Thank you so much for giving a useful library for REST API. But i have face one problem with your REST API, i have described below whole problem.

I have created website using Codeigniter 2.0 0 and its working fine. Now i am trying to apply REST in my web so i have downloaded codeigniter 3.0 and i have apply all require configuration for upgrade.

My system works fine in codeignier 3.0. Right now i have created only simple login module in 3.0, Now i am trying to convert login module for support REST API. But after added REST server in my controller i am not getting sessions. Here i am stuck why my session is lost when add REST server ?  i have write code in controller as  per explain in Tutorial.

Please help me to solve this problem.

I have also ask question in codeigniter forum for conflict codegniter 3.0  session with REST API.

http://forum.codeigniter.com/thread-61896-post-318632.html#pid318632

http://forum.codeigniter.com/thread-61878.html

**Comment by chriskacerguis** [comment]
Hello,

Some things I noticed.  

1.  It's on an error, it's a notice.

2.  Based on what I saw on the forum you are trying to access a key that doesn't exist (based on how you are setting it.

Regardless, it looks to be either an issue with CI 3 or your code not working with the changes in CI 3.

**Comment by hkachhia** [comment]
Yes its notice 

But if you have read whole thread properly, then you can find set session code.

$data_session_set = array('id' => $this->details->emp_reg_id, 'username' => $this->details->username, 'emp_role' => $this->details->emp_role, 'isLoggedIn' => 1);

$this->session->set_userdata($data_session_set);

For  your information this is working fine without REST library. 

Its displaying notice when i am used REST so i  have assign issue here.

I did not find any tutorial or example of REST with Session. So as per my knowledge I have wrote code in CI3. 

If you wants my code for testing, I will provide you two pages

**Comment by dschallert** [comment]
I'm experiencing the same issue..  Could it be that this is the case because of having the API on a subdomain, eq. api.mycorp.com and the request is issued from another domain eg. backend.mycorp.com ?

At first I thought it could be a issue with cookie domain, but i have set it to .mycorp.com , so i see no issue with sessions nor cookies directly.

**Comment by gravitymedianet** [comment]
I had the same issue. I was going crazy trying to find the answer. Then I realized that the session cookie was not being sent. Here is how I fixed it:

1. In my javascript which is making the api call using fetch api, I added:

`credentials: 'include'`

To the requestBody.

Then in my REST_Controller rest.php config file I added on line 635:

```
$config['forced_cors_headers'] = [
  'Access-Control-Allow-Credentials' => 'true'
];
```

And on line 565 added this to the array:

```
'Access-Control-Allow-Credentials'
```

Now, my PHP sessionid is being sent via cookie and my sessions are now working.
